File: sdks/python/souris_dw/exceptions.py
"""Exceptions for SourisDW."""


class SourisError(Exception):
    """Base exception for SourisDW."""
    pass


class DependencyError(SourisError):
    """Error with dependencies (yt-dlp, ffmpeg)."""
    pass


class DownloadError(SourisError):
    """Error during download."""
    pass


class ConfigError(SourisError):
    """Configuration error."""
    pass


class PlatformError(SourisError):
    """Unsupported platform error."""
    pass

File: sdks/nodejs/src/exceptions.js
/**
 * Exceptions for SourisDW.
 */

class SourisError extends Error {
  constructor(message) {
    super(message);
    this.name = 'SourisError';
  }
}

class DependencyError extends SourisError {
  constructor(message) {
    super(message);
    this.name = 'DependencyError';
  }
}

class DownloadError extends SourisError {
  constructor(message) {
    super(message);
    this.name = 'DownloadError';
  }
}

class ConfigError extends SourisError {
  constructor(message) {
    super(message);
    this.name = 'ConfigError';
  }
}

module.exports = { SourisError, DependencyError, DownloadError, ConfigError };
